library/core/src/intrinsics.rs

```diff
@@ -2666,12 +2666,17 @@ extern "rust-intrinsic" {
     ///
     /// `catch_fn` must not unwind.
     ///
-    /// The third argument is a function called if an unwind occurs (both Rust unwinds and foreign
-    /// unwinds). This function takes the data pointer and a pointer to the target-specific
-    /// exception object that was caught. For more information, see the compiler's source as well as
-    /// std's `catch_unwind` implementation.
+    /// The third argument is a function called if an unwind occurs (both Rust `panic` and foreign
+    /// unwinds). This function takes the data pointer and a pointer to the target- and
+    /// runtime-specific exception object that was caught.
     ///
-    /// The stable version of this intrinsic is `std::panic::catch_unwind`.
+    /// Note that in the case of a foreign unwinding operation, the exception object data may not be
+    /// safely usable from Rust, and should not be directly exposed via the standard library. To
+    /// prevent unsafe access, the library implementation may either abort the process or present an
```

Comment:
```suggestion
    /// prevent unsafe access, the standard library implementation may either abort the process or present an
```

maybe this is pedantic, but on my initial reading, I wasn't clear if "the library implementation" was (somehow) talking about the foreign library being invoked, or the Rust standard library.

(I freely admit that I also was not sure what it would mean for a foreign library to "make that choice", apart from the obvious option of just directly aborting the process, which I guess is always an option regardless of whatever else is going on.)

Reading the conversation associated with the next line (2676), I have inferred that the phrase "library implementation" is meant to refer to the Rust standard library.

So I figured I might suggest making that a tiny bit clearer.